src/mame/namco/namcos1_m.cpp

```diff
@@ -996,9 +1016,12 @@ u8 namcos1_state::faceoff_inputs_r(offs_t offset)
 		   much time reading the inputs and won't have enough cycles to play two
 		   digital sounds at once. This value is enough to read all inputs at least
 		   once per frame */
-		if (++m_strobe_count > 8)
+		if (!machine().side_effects_disabled())
+			++m_strobe_count;
+		if (m_strobe_count > 8)
 		{
-			m_strobe_count = 0;
+			if (!machine().side_effects_disabled())
+				m_strobe_count = 0;
 
 			res |= m_input_count;
 
```

Comment:
The code immediately below this sets `m_stored_input` – isn’t that going to be a side effect you don’t want triggered by debugger accesses?

Reply:
Fix side effect via https://github.com/mamedev/mame/pull/12962/commits/c114f4ac1c47f926bff57c636d37821595e50a72 .